qiskit/quantum_info/operators/symplectic/pauli_list.py

```diff
@@ -177,19 +181,22 @@ def _from_paulis(data):
         if not isinstance(data, (list, tuple, set, np.ndarray)):
             data = [data]
         num_paulis = len(data)
-        if num_paulis == 0:
-            raise QiskitError("Input Pauli list is empty.")
         paulis = []
         for i in data:
             if not isinstance(i, Pauli):
                 paulis.append(Pauli(i))
             else:
                 paulis.append(i)
-        num_qubits = paulis[0].num_qubits
+        if num_qubits is None:
+            if num_paulis == 0:
+                raise QiskitError("Input Pauli list is empty; cannot infer num_qubits.")
```

Comment:
minor message nit: could we say something like "pass the `num_qubits` argument to disambiguate" to tell the user how to fix the issue?